primitives/tests/advz.rs

```diff
@@ -32,7 +31,13 @@ fn round_trip() {
         );
 
     vid::round_trip(
-        |payload_chunk_size, num_storage_nodes| {
+        |payload_chunk_size, num_storage_nodes, multiplicity| {
+            let mut srs_rng = jf_utils::test_rng();
+            let srs = UnivariateKzgPCS::<Bls12_381>::gen_srs_for_testing(
+                &mut srs_rng,
+                num_coeff * multiplicity,
+            )
+            .unwrap();
```

Comment:
You don't need to move this inside the VID factory. Instead just do it once at the beginning (like before) but set it to `supported_degree * max_multiplicity` where `max_multiplicity` is the maximum possible multiplicity, something like
```rust
let max_multiplicity = multiplicities.iter().max();
```
This might speed up the test a bit.